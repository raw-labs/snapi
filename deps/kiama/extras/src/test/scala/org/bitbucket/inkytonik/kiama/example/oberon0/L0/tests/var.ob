MODULE var;

VAR
    avar1 : INTEGER;
    avar2 : avar1;
    avar4 : INTEGER;
    avar3 : INTEGER;
    avar3 : BOOLEAN;
    avar4 : INTEGER;
    avar5, avar5 : INTEGER;
    avar6 : avar3;
    avar7 : avar1;

BEGIN
    avar := 42;
    avar := bvar;
    avar := 99 - (avar + 1);
    
    avar1 := 1;
    avar2 := 1;
    avar3 := 1;
    avar4 := 1;
    avar5 := 1;
    avar6 := 1;
    avar7 := 1
END var.
