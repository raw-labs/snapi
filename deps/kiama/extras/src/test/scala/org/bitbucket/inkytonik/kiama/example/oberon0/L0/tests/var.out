* _pp.ob
MODULE var;

    VAR
        avar1 : INTEGER;
        avar2 : avar1;
        avar4 : INTEGER;
        avar3 : INTEGER;
        avar3 : BOOLEAN;
        avar4 : INTEGER;
        avar5, avar5 : INTEGER;
        avar6 : avar3;
        avar7 : avar1;

BEGIN
    avar := 42;
    avar := bvar;
    avar := 99 - (avar + 1);
    avar1 := 1;
    avar2 := 1;
    avar3 := 1;
    avar4 := 1;
    avar5 := 1;
    avar6 := 1;
    avar7 := 1
END var.
* errors
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:5:13:error: avar1 is not a type name
    avar2 : avar1;
            ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:8:5:error: avar3 is already declared
    avar3 : BOOLEAN;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:9:5:error: avar4 is already declared
    avar4 : INTEGER;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:10:12:error: avar5 is already declared
    avar5, avar5 : INTEGER;
           ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:12:13:error: avar1 is not a type name
    avar7 : avar1;
            ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:15:5:error: avar is not declared
    avar := 42;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:16:5:error: avar is not declared
    avar := bvar;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:16:13:error: bvar is not declared
    avar := bvar;
            ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:17:5:error: avar is not declared
    avar := 99 - (avar + 1);
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/var.ob:17:19:error: avar is not declared
    avar := 99 - (avar + 1);
                  ^
